### PR TITLE
Os change

### PR DESCRIPTION
**Pull request by BenediktMKuehne**
❗  EMBArk is changing to Ubuntu:jammy (Ubuntu 22 LTS).
To make containerization and development easier, **Kali-Linux will no longer be the base OS for testing and development.**

**What kind of change does this PR introduce?**
---
* Python version changed from 3.9 to 3.10
* dependencies are customized for ubuntu:jammy
* server directory changed to /var/www
* other minor additions

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
---
* ⚠️ Python version changed to 3.10 : **pipenv must be updated/rebuild**
* ⚠️ reinstallation required
* ❗ ⚠️ *newer kali-linux versions no longer actively supported* ❗ 

**Other information**:
---
https://github.com/e-m-b-a/emba/pull/258

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,4 +1,4 @@
-[submodule "EMBA"]
+[submodule "emba"]
 	path = emba
 	url = https://github.com/e-m-b-a/emba.git
-	branch = master
+    branch = os-change
```
Comment (m-1-k-3) [inline .gitmodules]: master

File: Dockerfile [hunk 2]
```diff
@@ -20,8 +20,10 @@ RUN apt-get update && apt-get -y -q --no-install-recommends install wget \
     pipenv \
     && rm -rf /var/lib/apt/lists/*
 
-RUN mkdir /app
+RUN mkdir /var/www
```
Comment (m-1-k-3) [inline Dockerfile]: currently we do not run with docker. Should we remove the docker stuff until it is finished?

File: embark/entrypoint.sh [hunk 3]
```diff
@@ -1,23 +1,65 @@
 #!/bin/bash
 
-export DJANGO_SETTINGS_MODULE=embark.settings.docker # TODO
+export DJANGO_SETTINGS_MODULE=embark.settings.docker
+export EMBARK_DEBUG=True
```
Comment (m-1-k-3) [inline embark/entrypoint.sh:4]: in bash True is true and this is 0. Could we just use 0?

File: embark/entrypoint.sh [hunk 4]
```diff
@@ -1,23 +1,65 @@
...
-python3 manage.py runapscheduler --test | tee -a ./logs/migration.log &
-echo -e "[*] Starting uwsgi - log to ./logs/uwsgi.log"
-uwsgi --wsgi-file /app/embark/embark/wsgi.py --http :80 --processes 2 --threads 10 --logto ./logs/uwsgi.log &
-echo -e "[*] Starting daphne - log to ./logs/daphne.log"
-# shellcheck disable=2094
-#TODO the bind address has to be set at runtime or dns-res
-daphne -v 3 --access-log ./logs/daphne.log embark.asgi:application -p 8001 -b '0.0.0.0' &> ./logs/daphne.log
+# db_init
+echo -e "\n[""$BLUE JOB""$NC""] Starting migrations - log to embark/logs/migration.log"
+python3.10 manage.py makemigrations users uploader dashboard reporter | tee -a /var/www/logs/migration.log
```
Comment (m-1-k-3) [inline embark/entrypoint.sh:34]: could we define this as variable in the beginning. Then we can see at beginning of the script which Python version is used

File: embark/entrypoint.sh [hunk 5]
```diff
@@ -1,23 +1,65 @@
...
+python3.10 manage.py migrate | tee -a /var/www/logs/migration.log
```
Comment (m-1-k-3) [inline embark/entrypoint.sh:35]: same

File: entrypoint.sh [hunk 6]
```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+
+export DJANGO_SETTINGS_MODULE=embark.settings.docker
+export EMBARK_DEBUG=True
```
Comment (m-1-k-3) [inline entrypoint.sh]: =0 possible?

File: entrypoint.sh [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+export HTTP_PORT=80
+export HTTPS_PORT=443
+export BIND_IP='0.0.0.0'
+export FILE_SIZE=2000000000
+
+# check emba
```
Comment (m-1-k-3) [inline entrypoint.sh]: EMBA

File: entrypoint.sh [hunk 8]
```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+
+export DJANGO_SETTINGS_MODULE=embark.settings.docker
+export EMBARK_DEBUG=True
+export HTTP_PORT=80
+export HTTPS_PORT=443
+export BIND_IP='0.0.0.0'
+export FILE_SIZE=2000000000
+
```
Comment (m-1-k-3) [inline entrypoint.sh]: Should we also define /app/www somewhere. In the future we just need to change it once

File: installer.sh [hunk 9]
```diff
@@ -90,42 +93,53 @@ write_env() {
...
-  # create server sign requests (csr)
-  openssl genrsa -out embark.local.key 2048
-  openssl req -new -sha256 -key embark.local.key -out embark.local.csr  -subj '/CN=embark.local/O=EMBA/C=US'
-  openssl genrsa -out embark-ws.local.key 2048
-  openssl req -new -sha256 -key embark-ws.local.key -out embark-ws.local.csr  -subj '/CN=embark-ws.local/O=EMBA/C=US'
-  # signe csr with ca
-  openssl x509 -req -in embark.local.csr -CA rootCA.crt -CAkey rootCA.key -CAcreateserial -out embark.local.crt -days 10000 -sha256
-  openssl x509 -req -in embark-ws.local.csr -CA rootCA.crt -CAkey rootCA.key -CAcreateserial -out embark-ws.local.crt -days 10000 -sha256
+  if [[ -f embark.local.csr ]] || [[ -f embark-ws.local.csr ]] || [[ -f embark.local.crt ]] || [[ -f embark-ws.local.crt ]]; then 
+    echo -e "\n$GREEN""$BOLD""Certs already generated, scipping""$NC"
```
Comment (m-1-k-3) [inline installer.sh]: typo ... skipping

File: installer.sh [hunk 10]
```diff
@@ -206,69 +220,103 @@ reset_docker() {
...
   #add dns name
   dns_resolve
 
   #install packages
-  PIPENV_VENV_IN_PROJECT=1 pipenv install
+  cp ./Pipfile* /var/www/
+  (cd /var/www && PIPENV_VENV_IN_PROJECT=1 pipenv install)
+  
+  # set secret-key
+  DJANGO_SECRET_KEY=$(python3 -c 'from django.core.management.utils import get_random_secret_key; print(get_random_secret_key())')
```
Comment (m-1-k-3) [inline installer.sh]: python3 vs python3.10

File: installer.sh [hunk 11]
```diff
@@ -422,61 +478,90 @@ install_embark_docker(){
 }
 
 uninstall (){
-  echo -e "$ORANGE""$BOLD""Deleting Configuration and reseting""$NC"
+  echo -e "[+]$CYAN""$BOLD""Deleting Configuration and reseting""$NC"
```
Comment (m-1-k-3) [inline installer.sh]: Configuration and reset?

File: installer.sh [hunk 12]
```diff
@@ -422,61 +478,90 @@ install_embark_docker(){
...
 
-  #1 delete symlink
+  # delete symlink (lagacy)
```
Comment (m-1-k-3) [inline installer.sh]: legacy vs lagacy

File: installer.sh [hunk 13]
```diff
@@ -422,61 +478,90 @@ install_embark_docker(){
...
-  rm /app
+  if ! [[ -f /app ]]; then
+    if [[ $( readlink /app ) == "$PWD" ]]; then
+      rm /app
+    fi
+  fi
 
-  #2 delete www and upload-dir
+  # delete dirs
   echo -e "$ORANGE""$BOLD""Delete Directories""$NC"
```
Comment (m-1-k-3) [inline installer.sh:492]: directories ->d

**Comment by m-1-k-3** [comment]
looks good. Please check the notes. I will do another functionality check and then we are set good to go

**Comment by torabi12** [comment]
Hello Benedikt,

I tried the EMBArk installation on Ubuntu 22.04 LTS, it was a new installaton and I manually installed python3.9 before I started the EMBArk installer with -d.

During the installation I got some errors, please see the attachments:
kép

kép

kép

I am not sure, maybe I tried too early the installation on Ubuntu, but I wanted to share with you my experiences.

Best regards,
torabi

**Comment by BenediktMKuehne** [comment]
Hi @torabi12 ,

I'm guessing you tried the embark:main branch?
Once this PR is merged the installation EMBArk should work as expected.
Sorry 😅
Switching to ubuntu was more complicated than expected

**Comment by torabi12** [comment]
Hello Benedikt,

I installed Ubuntu 22.04 LTS with default parameters in VMWare Workstation and I added python3.9 with these comands:
$ sudo apt update
$ sudo apt install software-properties-common
$ sudo add-apt-repository ppa:deadsnakes/ppa
$ sudo apt install python3.9

and the basic EMBArk installation came:
$ git clone https://github.com/e-m-b-a/embark.git
$ cd embark
$ sudo ./installer.sh -d

Without python3.9 I got other errors but now I don't know how to fix the actual issues:)
If python3.9 is needed for EMBArk can you put a verification into the installer and download it automatically within EMBArk installation? I thin Ubuntu does not contain this by default.

If I can try another branch please let me know how to do it and I will try another installation as well.

Regards,
torabi

**Comment by BenediktMKuehne** [comment]
@torabi12 
`git clone https://github.com/BenediktMKuehne/embark -b os-change`

Feel free to note any issue you encounter.

**Review by m-1-k-3 — APPROVED** [review]
Great work! Thank you

**Comment by torabi12** [comment]
Hello Benedikt,

I installed this morning EMBArk using your source:
https://github.com/BenediktMKuehne/embark

there were some minor warnings during the instalation, please verify them which one is valid or needs modification. I insert the images like in the command line from up to down starting from IF20_cve_search part:

kép

kép

kép

kép
This is displayed 3 times in this section.

And the last one:
kép

When I started EMBArk I got this:
kép

and this:
kép

I did a short scan with EMBArk to see the basic functions:
kép

but on the report page the firmware version ignores the dots. Under that the device skipped the spaces and the notes are also:
kép

I think in the report is used to be identical but now the report sometimes cuts out parts which is not user friendly.
The version number would be the most important because it always contain . or - and sometimes other character and it can be long and if we skip these the reading is not comfortable.

Firmware delete was working fine. :)

Regards
torabi

**Comment by BenediktMKuehne** [comment]
Hi @torabi12,
at this point I'm not bothering with pip-warnings to much, as long as the modules work. (There is to much changing right now)
There apparently is also a general issue with python-pip and the setup.py, which is unrelated to EMBArk.

The dbus-error could be related to EMBAs new status-bar.
Looks like a bug
Are you on a desktop or a server-version of ubuntu 22.04 LTS ?
If you could open a issue in EMBA that'd be helpful.

Regarding the report:
Yes, currently user-input gets filtered.
In the future I'll remove the EMBA-flags with direct user-input completely.
Reason being unfiltered user-input (direct command-injection) was possible.
I know it's a bit annoying right now. Sorry 🤷

Thanks for the feedback

**Comment by torabi12** [comment]
Thank you, I opened a bug for EMBA:  dbus-launch error #284
I am using Ubuntu desktop.

**Comment by torabi12** [comment]
I also checked the main branch:
git clone https://github.com/e-m-b-a/embark.git

The error you have mentioned I think came:
kép

and the installation stopped at the point:
kép

Looks like the installation is terminated.

**Comment by torabi12** [comment]
I tried the main branch also and the installation went through but the EMBArk start looks bad:
kép

during the installation I think you have mentioned this error already:
kép

**Pull request merged**